official/utils/logs/logger.py

```diff
@@ -179,14 +181,13 @@ def log_metric(self, name, value, unit=None, global_step=None, extras=None):
     """
     metric = _process_metric_to_json(name, value, unit, global_step, extras)
     if metric:
-      with tf.gfile.GFile(
-          os.path.join(self._logging_dir, METRIC_LOG_FILE_NAME), "a") as f:
-        try:
-          json.dump(metric, f)
-          f.write("\n")
-        except (TypeError, ValueError) as e:
-          tf.logging.warning("Failed to dump metric to log file: "
-                             "name %s, value %s, error %s", name, value, e)
+      try:
+        json.dump(metric, self._metric_file_fh)
+        self._metric_file_fh.write("\n")
+        self._metric_file_fh.flush()
```

Comment:
Flush should probably be in a finally block? Realistically, it's unlikely to matter, I guess. Your call.